src/i18n/i18n-setup.ts

```diff
@@ -0,0 +1,46 @@
+import Vue from 'vue';
+import VueI18n from 'vue-i18n';
+import { Cookie } from '@nimiq/utils';
+
+Vue.use(VueI18n);
+
+const DEFAULT_LANGUAGE = 'en';
+const SUPPORTED_LANGUAGES = [DEFAULT_LANGUAGE];
+const LOADED_LANGUAGES: string[] = []; // our default language that is preloaded, if any
+
+export const i18n = new VueI18n({
+    locale: DEFAULT_LANGUAGE, // set locale (2 letters format: 'en')
+    fallbackLocale: DEFAULT_LANGUAGE, // fallback locale if no translation found
+    silentTranslationWarn: true, // disable the "no translation found" warning
+});
+
+function setI18nLanguage(lang: string) {
+    i18n.locale = lang;
```

Comment:
The hub does support to change the language. Just calling `loadLanguageAsync` should change the language and update translations once the language is loaded without having to reload the hub. (but the language needs to be included in the `SUPPORTED_LANGUAGES` array).

Reply:
Sure, sure, the hub itself basically supports changing the language. I was rather saying there is no way to change the language from within the hub yet. But nonetheless the hub should already be able to correctly switch the language.
Currently, on language change, not everything is re-translated, namely things that are translated outside of templates, i.e. translations made in the javascript code. Therefore I was suggesting to add a reload here.

Reply:
oh yes you're right. And since not everything can be turned into a getter, we need to reload components that use non-reactive translations.
But is there a way to reload a component without loosing the current state of this one ? Otherwise we need to tell the user that the selected language changed and that some part of the page (or the entire page) need to be reloaded in order to show all the correct translations. And if the user doesn't want to reload because he's in the middle of something, then we don't.
So, on tab focus, if the language changed, i ask through a `confirm` (or smth like that) if the user want to reload the page to update the translations and i warn him that whatever he was currently doing will be lost. Is it ok for you ?

Reply:
In my opinion it's fine to just reload the page but you could ask in our slack for more opinions.